File: app/src/main/java/org/ole/planet/myplanet/utilities/KeyboardUtils.java
package org.ole.planet.myplanet.utilities;

import android.app.Activity;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.view.View.OnTouchListener;

public class KeyboardUtils {

    public static void hideSoftKeyboard(Activity activity) {
        try {
            InputMethodManager inputMethodManager =
                    (InputMethodManager) activity.getSystemService(
                            Activity.INPUT_METHOD_SERVICE);
            inputMethodManager.hideSoftInputFromWindow(
                    activity.getCurrentFocus().getWindowToken(), 0);
        }catch (Exception e){}
    }

    public static void showSoftKeyboard(Activity activity) {
        InputMethodManager inputMethodManager =
                (InputMethodManager) activity.getSystemService(
                        Activity.INPUT_METHOD_SERVICE);
        inputMethodManager.showSoftInputFromInputMethod(
                activity.getCurrentFocus().getWindowToken(), 0);
    }

    public static void setupUI(View v, Activity activity) {
        // Set up touch listener for non-text box views to hide keyboard.
        OnTouchListener onTouchListener= new OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                hideSoftKeyboard(activity);
                return false;
            }
        };

        if (!(v instanceof EditText)) {
            v.setOnTouchListener(onTouchListener);
        }

        //If a layout container, iterate over children and seed recursion.
        if (v instanceof ViewGroup) {
            ViewGroup vg = (ViewGroup)v;
            for (int i = 0; i < vg.getChildCount(); i++) {
                View innerView = vg.getChildAt(i);
                setupUI(innerView, activity);
            }
        }

    }
}

File: app/src/main/java/org/ole/planet/myplanet/utilities/KeyboardUtils.kt
package org.ole.planet.myplanet.utilities

import android.app.Activity
import android.view.MotionEvent
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.EditText

object KeyboardUtils {
    fun hideSoftKeyboard(activity: Activity) {
        try {
            val inputMethodManager = activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
            inputMethodManager.hideSoftInputFromWindow(activity.currentFocus?.windowToken, 0)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

//    fun showSoftKeyboard(activity: Activity) {
//        val inputMethodManager =
//            activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
//        inputMethodManager.showSoftInputFromInputMethod(activity.currentFocus!!.windowToken, 0)
//    }

    fun setupUI(v: View, activity: Activity) {
        // Set up touch listener for non-text box views to hide keyboard.
        val onTouchListener = View.OnTouchListener { _: View?, _: MotionEvent? ->
            hideSoftKeyboard(activity)
            false
        }
        if (v !is EditText) {
            v.setOnTouchListener(onTouchListener)
        }

        //If a layout container, iterate over children and seed recursion.
        if (v is ViewGroup) {
            for (i in 0 until v.childCount) {
                val innerView = v.getChildAt(i)
                setupUI(innerView, activity)
            }
        }
    }
}
